x-pack/legacy/plugins/lens/public/xy_visualization_plugin/xy_expression.tsx

```diff
@@ -291,3 +282,40 @@ export function XYChart({ data, args, formatFactory, timeZone }: XYChartRenderPr
     </Chart>
   );
 }
+
+/**
+ * Renames the columns to match the user-configured accessors in
+ * columnToLabelMap. If a splitAccessor is provided, formats the
+ * values in that column.
+ */
+function sanitizeRows({
+  splitAccessor,
+  table,
+  formatFactory,
+  columnToLabelMap,
+}: {
+  splitAccessor?: string;
+  table: KibanaDatatable;
+  formatFactory: FormatFactory;
+  columnToLabelMap: Record<string, string | undefined>;
+}) {
+  const column = table.columns.find(c => c.id === splitAccessor);
+  const formatter = formatFactory(column && column.formatHint);
+
+  return {
+    splitAccessor: column && column.id,
+    rows: table.rows.map(r => {
+      const newRow: typeof r = {};
+
+      if (column) {
+        newRow[column.id] = formatter.convert(r[column.id]);
+      }
+
+      Object.keys(r).forEach(key => {
+        const newKey = columnToLabelMap[key] || key;
+        newRow[newKey] = r[key];
+      });
+      return newRow;
+    }),
+  };
```

Comment:
So you've changed the row behavior here, it's no longer 1:1 with the data in?

Reply:
Rows are 1:1, but we're adding a column. That way the mapped column can still be used on the X or Y axes if need be.